### PR TITLE
[test-runner] Enable IOS RenderTestRunner and run it on Firebase

### PR DESCRIPTION
**Pull request by zmiao**
This PR will enable running mbgl-render-test-runner on ios devices.

### CODE REVIEW

File: next/platform/ios/ios.cmake [hunk 1]
```diff
@@ -9,8 +9,8 @@ endif()
 
 macro(initialize_ios_target target)
     set_target_properties(${target} PROPERTIES XCODE_ATTRIBUTE_IPHONEOS_DEPLOYMENT_TARGET "${IOS_DEPLOYMENT_TARGET}")
-    set_target_properties(${target} PROPERTIES XCODE_ATTRIBUTE_ENABLE_BITCODE "YES")
-    set_target_properties(${target} PROPERTIES XCODE_ATTRIBUTE_BITCODE_GENERATION_MODE bitcode)
+    # set_target_properties(${target} PROPERTIES XCODE_ATTRIBUTE_ENABLE_BITCODE "YES") set_target_properties(${target} PROPERTIES
```
Comment (alexshalamov) [inline next/platform/ios/ios.cmake]: why these are disabled?

File: metrics/next-ios-render-test-runner-style.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
+    "base_test_path": "integration",
+    "cache_path": "cache-style.db",
+    "expectation_paths": [
+        "expectations/platform-all"
+    ],
+    "ignore_paths": [
+        "ignores/platform-all.json",
+        "ignores/platform-macos.json"
+    ]
```
Comment (alexshalamov) [inline metrics/next-ios-render-test-runner-style.json]: Metrics would be added later?

File: circle.yml [hunk 3]
```diff
@@ -671,6 +679,87 @@ commands:
...
+          when: always
+          command: |
+            if [[ -n "${GCLOUD_SERVICE_ACCOUNT_JSON_IOS}" ]]; then
+              testResult=$(gsutil ls -d gs://test-lab-hn7kwqcz3108z-hjm4qtbqqkymk/ios-render-test-app-${CIRCLE_BUILD_NUM}/iphonexs-12.1-en-portrait/TestLogs/)
+              mkdir -p render-test-app
+              gsutil -m cp -r $testResult render-test-app
+              cd render-test-app/TestLogs && ls
+              xcparse attachments *.xcresult
+              cd ../.. && mkdir -p /tmp/tests/ios-render-test
+              cp render-test-app/TestLogs/*.html /tmp/tests/ios-render-test
```
Comment (alexshalamov) [inline circle.yml]: Metrics would be added in follow-up prs?

File: render-test/ios/fastlane/README.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+Description of what the lane does
+### ios ci_build
+```
+fastlane ios ci_build
+```
+Description of what the lane does
+
+----
+
+This README.md is auto-generated and will be re-generated every time [fastlane](https://fastlane.tools) is run.
```
Comment (alexshalamov) [inline render-test/ios/fastlane/README.md]: should it be added to gitignore?

File: render-test/ios/setup_test_data.sh [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+rm -rf ../../test-data
+mkdir ../../test-data
+mkdir ../../test-data/integration
+
+cp -r ../../mapbox-gl-js/test/integration/render-tests ../../test-data/integration/
+cp -r ../../mapbox-gl-js/test/integration/query-tests ../../test-data/integration/
+cp -r ../../metrics/expectations ../../test-data/
+cp -r ../../metrics/ignores ../../test-data/
+# cp -r ../../metrics/tests/
+# cp -r ../../metrics/next-android-render-test-runner/
```
Comment (alexshalamov) [inline render-test/ios/setup_test_data.sh]: Still needed?

File: next/platform/ios/ios.cmake [hunk 6]
```diff
@@ -89,4 +92,89 @@ target_link_libraries(
         z
 )
 
+if(IOS_RENDER_TESTING)
```
Comment (alexshalamov) [inline next/platform/ios/ios.cmake]: `CMAKE_SYSTEM_NAME STREQUAL iOS`?

File: next/CMakeLists.txt [hunk 7]
```diff
@@ -22,6 +22,7 @@ option(MBGL_WITH_CORE_ONLY "Build only the core bits, no platform code" OFF)
 option(MBGL_WITH_COVERAGE "Enable code coverage collection" OFF)
 option(MBGL_WITH_QT "Build Mapbox GL Qt bindings" OFF)
 option(MBGL_WITH_SANITIZER "Use [address|thread|undefined] here" OFF)
+option(IOS_RENDER_TESTING "Enable ios render test runner" OFF)
```
Comment (alexshalamov) [inline next/CMakeLists.txt]: why is this needed can we create create runner target for ios automatically?

**Comment by jasondowalter** [comment]
I have no clue what to do lol

File: metrics/next-ios-render-test-runner-style.json (hunk 2 above)
Comment (zmiao) [reply metrics/next-ios-render-test-runner-style.json]: Yes, currently there are several failing test cases, the following steps would be making all tests passing(adding ios ignores, expectations), generating metrics.

File: circle.yml (hunk 3 above)
Comment (zmiao) [reply circle.yml]: yes!

File: next/platform/ios/ios.cmake (hunk 6 above)
Comment (zmiao) [reply next/platform/ios/ios.cmake]: Currently we have another job next-ios-xcode-release, which also has ios platform.

File: next/platform/ios/ios.cmake (hunk 1 above)
Comment (zmiao) [reply next/platform/ios/ios.cmake]: Without commenting these two lines, xcodebuild will always fail, need help from @tmpsantos

Comment (zmiao) [reply next/platform/ios/ios.cmake]: Fixed.

File: render-test/ios/setup_test_data.sh (hunk 5 above)
Comment (zmiao) [reply render-test/ios/setup_test_data.sh]: Removed.

File: next/CMakeLists.txt (hunk 7 above)
Comment (zmiao) [reply next/CMakeLists.txt]: @tmpsantos need you help!

File: metrics/next-ios-render-test-runner-style.json (hunk 2 above)
Comment (zmiao) [reply metrics/next-ios-render-test-runner-style.json]: Now all tests are passed, platform dependent ignores and expectations are added. Metrics will be done in follow-up pr.
